liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -870,14 +905,17 @@ public boolean recompileBuildFile(File buildFile, Set<String> compileArtifactPat
                     util.restartServer();
                     return true;
                 } else {
-                    // TODO: confirm that a call to generate features is required when a build file is modified 
-                    // (ie. changes are not picked up by class file changes)
-                    // if we generate features here, we will also need to clear javaSourceClasses
-                    // we will also need to skip installing features on a failure
+                    // TODO: confirm that a call to generate features is required when a build file's compilation dependencies are modified
+                    // if we generate features here we will also need to skip installing features on
+                    // a failure
                     if (compileDependenciesChanged && generateFeatures) {
                         // build file change - provide updated classes and all existing features to binary scanner
                         Collection<String> javaSourceClassPaths = util.getJavaSourceClassPaths();
-                        libertyGenerateFeatures(javaSourceClassPaths, false);
+                        if (!javaSourceClassPaths.isEmpty()) { // only call generateFeatures if there are updated classes to generate for
+                            if (libertyGenerateFeatures(javaSourceClassPaths, false)) {
```

Comment:
For readability and future maintenance, it would be better to do the method call and then boolean check in two steps.